zulipterminal/ui_tools/boxes.py

```diff
@@ -476,14 +476,11 @@ def soup2markup(self, soup: Any, **state: Any) -> List[Any]:
                     # Prepend org url to convert it to an absolute link
                     link = urljoin(self.model.server_url, link)
 
-                if link == text:
-                    # If the link and text are same
-                    # usually the case when user just pastes
-                    # a link then just display the link
-                    markup.append(('msg_link', text))
-                else:
-                    markup.append(
-                        ('msg_link', '[' + text + ']' + '(' + link + ')'))
+                text = text if text else link
```

Comment:
I was wondering if it would be better to trim the links that do not have any text to improve the flow of a message? We'd show the complete link in the footnote anyway.

Reply:
Yes, I meant the last segment. I have applied it in the latest update.